### PR TITLE
Added: dark-mode to Graphs/Charts

### PR DESCRIPTION
**Pull request by rishav4101**
Description : Graphs and Charts in dark-mode.
Signed off by : @rishav4101 rajkumarrishav4101@gmail.com
Fixes : #386 
Proof :
ano
darkgraph

### CODE REVIEW

**Comment by rishav4101** [comment]
@Harkishen-Singh @muskankhedia Please review it.

File: dashboard/v1.1/src/pages/Dashboard/CPUUsage.tsx [hunk 1]
```diff
@@ -54,6 +53,9 @@ const CPUUsage: FC<CPUUsageProps> = ({ cpuMetrics }) => {
         opacityFrom: 1,
         opacityTo: 0.2
       }
+    },
+    theme: {
+      mode: localStorage.getItem('dark-mode') === 'true' ? 'dark' : 'light'
```
Comment (aquibbaig) [inline dashboard/v1.1/src/pages/Dashboard/CPUUsage.tsx:58]: Let's not reference localStorage in every case, what I'd say is to use localStorage once(if you can) in any top-level component and pass it as a context variable. The issue is that the read performance of localStorage can be costly at times. I've noticed my browser freezing on this branch and referencing localStorage constantly can be a cause. Also if all the charts have the same theme prop, what about instantiating it once and referencing it everywhere.

**Review by aquibbaig — CHANGES_REQUESTED** [review]
Hey, thanks for this PR! I can see that my browser freezes for a couple of seconds when I toggle dark mode, (dark mode should be seamless). There are a couple of reasons why it may happen and how we can fix it.
## Multiple references of localStorage in our code
 - jsperf says that the read performance of `localStorage` is significantly less than `native`. It may have less effect on real world uses but even if it doesn't, we must look for best practices here.
 - The solution here is to gather all references of theme variable and create a context variable in any top-level component and pass this as a prop to page component where you can say `const { darkMode } = useContext(Theme)`. 
## Multiple re-renders of charts cause the page to freeze.
- Since on a dark mode, we re-render charts with all their data in memory, this can be a costly operation.
- The solution here is to use code-splitting and use `React.Suspense` and `React.lazy` to show loaders until the charts are loaded and display them after that.

**Comment by muskankhedia** [comment]
@rishav4101 Can you please fix this?

Screenshot 2021-03-20 at 2 37 22 PM

**Comment by rishav4101** [comment]
@muskankhedia @aquibbaig Kindly review it.

**Comment by muskankhedia** [comment]
@rishav4101 Can you please squash all the commits to one commit?

**Review by muskankhedia — CHANGES_REQUESTED** [review]
Can you please fix the graph on the `monitoring` and `tests` screen as well?

Screenshot 2021-03-22 at 11 24 51 AM

Screenshot 2021-03-22 at 11 25 55 AM

**Comment by muskankhedia** [comment]
@rishav4101 If you check the `Files changed` now, you could see the previously edited files have been edited in your commit. I will advise you to create a new PR only with your changes or revert the changes in the files that aren't a part of this PR. Can you please pick any one of the option and edit accordingly?

**Comment by rishav4101** [comment]
Yes, I will create a new PR for the change.

**Comment by rishav4101** [comment]
I have created a new PR #420, closing this one.

**Pull request closed without merging**